### PR TITLE
Re-add linker attributes to generic type params of JsonSerializer methods

### PR DESCRIPTION
**Pull request by layomia**
Missed in https://github.com/dotnet/runtime/pull/41484.

### CODE REVIEW

File: src/libraries/System.Text.Json/ref/System.Text.Json.cs [hunk 1]
```diff
@@ -188,18 +188,18 @@ public static partial class JsonSerializer
...
+        public static System.Threading.Tasks.ValueTask<TValue?> DeserializeAsync<[System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicConstructors | System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicFields | System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicProperties)] TValue>(System.IO.Stream utf8Json, System.Text.Json.JsonSerializerOptions? options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static TValue? Deserialize<[System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicConstructors | System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicFields | System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicProperties)] TValue>(System.ReadOnlySpan<byte> utf8Json, System.Text.Json.JsonSerializerOptions? options = null) { throw null; }
```
Comment (jkotas) [inline src/libraries/System.Text.Json/ref/System.Text.Json.cs:192]: Do we need the attributes in the implementation as well?

**Review by jkotas — APPROVED** [review]

**Comment by jkotas** [comment]
> Missed in #41484.

How was this missed? Is there a bug in the ref-file auto-generation in handling these?

**Comment by layomia** [comment]
> How was this missed? Is there a bug in the ref-file auto-generation in handling these?

Yes, generic type params are currently being ignored. Called out in https://github.com/dotnet/runtime/pull/41957#issuecomment-691741866. cc @safern on whether this is known/tracked.

**Comment by safern** [comment]
Yeah, that's a known issue: https://github.com/dotnet/arcade/issues/5925

**Pull request merged**